[phiphi-tan] iP

phiphi-tan opened the pull request:
# PHI :dragon_face: :rowboat:
>"Coding is a mental construct, just think more computing thoughts" - Me


*PHI* frees your mind of having to remember things you need to do. It's
- text-based
- easy to figure out
- encapsulates the very human touch of having somebody exasperated with your improper user inputs.

All you need to do is:
1. Download it from [here](https://github.com/phiphi-tan/ip)
2. Clone it (and stop complaining)
3. Do your stuff and watch it manage your tasks for you

Best part of all, it is completely **FREE** (~~unlike its creator's time~~)!

Look at all the things it can do:
- [x] Managing to-do items
- [x] Managing tasks with deadlines
- [x] Managing events
- [x] Search feature for keywords
- [ ] Reminders (coming soon, or not)
----
Since you're still here, here's a chunk of code for you to test out your coding abilities:
   ```java
   /*
   Welcome to
    ___ _  _ ___
   | _ \ || |_ _|
   |  _/ __ || | 
   |_| |_||_|___|
   PHI (Programmable Human Interface)
   */
 
public class Main {
    public static void main(String[] args) {
        System.out.println("You are a nerd");
    }  
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

phiphi-tan committed:
"Level-0 Commit"

phiphi-tan committed:
Level-1: Echo

phiphi-tan committed:
Level-1

phiphi-tan committed:
Level-2: Add, List

phiphi-tan committed:
Level-3. Mark as Done

phiphi-tan committed:
Level 4. ToDos, Events, Deadlines

phiphi-tan committed:
A-TextUiTesting: Automated Text UI Testing

phiphi-tan committed:
Level-5: Handle Errors

phiphi-tan committed:
Level-6: Delete

phiphi-tan committed:
A-Enums: Use Enums

phiphi-tan committed:
Added ./data/tasklist.txt and read/write functionality

phiphi-tan committed:
Added proper abstraction and support for FileNotFound exceptions

phiphi-tan committed:
Updated Deadline.java and Event.java for LocalDate handling

phiphi-tan committed:
Added Parser.java, Storage.java, Ui.java for more OOP abstractions

phiphi-tan committed:
Organised classes into a single package and fixed access modifiers

phiphi-tan committed:
Added gradle support

phiphi-tan committed:
Added JUnit tests and fixed code accordingly

phiphi-tan committed:
Updated code for .jar file

phiphi-tan committed:
Updated README.md as a project log

phiphi-tan committed:
Added JavaDoc comments to all classes

phiphi-tan committed:
Made slight changes to comply with coding standards

phiphi-tan committed:
Added basic search functionality

phiphi-tan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/phi/Storage.java
#	src/main/java/phi/TaskList.java

phiphi-tan committed:
Merge branch 'branch-Level-9'

phiphi-tan committed:
Updated README.md and cleaned up code

weeweh commented on src/main/java/phi/Phi.java:
```diff
@@ -0,0 +1,52 @@
+    private final Ui phiUi;
+    private final Storage taskStorage;
+    private final TaskList tasks;
+    private final Parser phiParser;
```
Why is phi in the name of the Ui and Parser?
Phi should have a Ui and Parser, not a phiUi and phiParser

weeweh commented on src/main/java/phi/Task.java:
```diff
@@ -0,0 +1,51 @@
+        this.taskName = taskName;
+    }
+
+    public void markDone() {
```
May consider a setDone(boolean done) instead.

weeweh commented on src/main/java/phi/Task.java:
```diff
@@ -0,0 +1,51 @@
+     */
+    public abstract String outputFormat();
+
+    public String getMsg() {
```
why is message abbreviated?

weeweh commented on src/main/java/phi/TaskList.java:
```diff
@@ -0,0 +1,174 @@
+     */
+    public String deleteTask(String input) {
+        try {
+            int number = Integer.parseInt(input.substring(7));
```
number may be too general. Index might be more suitable.

weeweh reviewed (commented):
Liked the README.md and the help command.
Minor name changes. Should consider breaking down the if-else blocks using a Command class.

eyelessrhyme7 commented on README.md:
```diff
@@ -1,24 +1,98 @@
-# Duke project template
+# CS2103T Individual Project (iP)
+
+This is a project log for my individual project
```
Great job on logging in your own progress on the README.md. Provides clear overview of milestones and helps review commits.

eyelessrhyme7 commented on src/main/java/phi/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public String outputFormat() {
```
Great job on isolating writing of output to file in a separate function. Perhaps, the name for the function can be a bit more explanative, indicating that it is used to return output for file writing, not to be confused with output for the chatbot itself. 🤔

eyelessrhyme7 commented on src/main/java/phi/Event.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
Consider adding documentation for the outputFormat function so it is clear that it produces output to write to file.

eyelessrhyme7 commented on src/main/java/phi/Parser.java:
```diff
@@ -0,0 +1,115 @@
+     * @param input User-input given to PHI
+     * @return      Response of the parser to the input, to be printed by PHI
+     */
+    public String handle(String input) {
```
Consider giving the function a more descriptive name, such as handleInput() or parseInput() to improve readability. 😄

eyelessrhyme7 commented on src/main/java/phi/Parser.java:
```diff
@@ -0,0 +1,115 @@
+                } else if (!input.contains("/from")) {
+                    return "You're missing a ' \"/from\" flag, halfwit";
+                } else if (!input.contains("/to")) {
+                    return "You're missing a \"/to\" flag, dimwit";
```
Suggest being nicer to the end-user, as in the todo, deadline cases above. Refrain from using hurtful language in user interactions. 😞

eyelessrhyme7 commented on src/main/java/phi/Phi.java:
```diff
@@ -0,0 +1,52 @@
+     * @param filePath the file path of the .txt file to be loaded from storage
+     */
+    public Phi(String filePath) {
+        phiUi = new Ui();
```
Consider adding this. keyword when initializing attributes for readability.

eyelessrhyme7 commented on src/main/java/phi/Phi.java:
```diff
@@ -0,0 +1,52 @@
+        phiParser = new Parser(tasks);
+    }
+
+    public static void main(String[] args) {
```
Consider repositioning the main function at the end, to improve readability and understanding.

eyelessrhyme7 commented on src/main/java/phi/TaskList.java:
```diff
@@ -0,0 +1,174 @@
+     * @param input User-inputted string to be parsed
+     * @return      String response
+     */
+    public String doTask(String input) {
```
markDone() method above and doTask(String) are performing similar actions and can be confusing. Consider renaming or re-organizing the computation elsewhere, keeping only one do function in Task class.

eyelessrhyme7 commented on src/main/java/phi/TaskList.java:
```diff
@@ -0,0 +1,174 @@
+        return output.toString();
+    }
+
+    public int listSize() {
```
Consider renaming to getSize() to name function as a verb

eyelessrhyme7 commented on src/main/java/phi/Ui.java:
```diff
@@ -0,0 +1,34 @@
+        System.out.println(exitMsg);
+    }
+
+    public static String helpMsg() {
```
Consider placing the content of helpMsg elsewhere as a constant. Same for greeting and goodbye functions above. Try to not use functions as storage.

eyelessrhyme7 reviewed (commented):
Appreciate and impressed by the UI improvement and detailed feedback in the interaction to the user. Consider changing a few function names, and abstracting away storing of strings in constants rather than functions.

phiphi-tan committed:
Updated README.md and added the changelog

phiphi-tan committed:
Implement checkStyle (with Gradle) and fix styling errors

phiphi-tan committed:
Add GUI using JavaFX

phiphi-tan committed:
Package classes further and tidy code

phiphi-tan committed:
Update build.gradle, README and changelog for new release

phiphi-tan committed:
Use assertions

Private methods in some classes have invariants for their arguments.

Such invariants should be checked to ensure assumptions made about
such arguments are right. They are typically to be tested at the
start of methods.

Assertions are used to check for logically impossible situations,
but they will not be used at runtime.

phiphi-tan committed:
Remove I/O redirection style automated testing and text-based UI

phiphi-tan committed:
Improve code quality

There are processes to follow when user inputs do not
fit expected parameters. In most cases, the error is thrown through
IllegalArgumentException.

Using a checked exception forces the case to be explicitly handled.

Defining and using a custom PhiException allows for greater control
as to what happens after unexpected user inputs occur.

phiphi-tan committed:
Merge branch 'master' into branch-A-Assertions

phiphi-tan committed:
Merge pull request #2 from phiphi-tan/branch-A-Assertions

Use assertions

phiphi-tan committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/phi/util/Parser.java
#	src/main/java/phi/util/TaskList.java

phiphi-tan committed:
Provide more help to users

Users only have access to a help command, which may be daunting
when first experiencing a new interface and app.

Pre-load the app with sample data at the first run, and polish the
help command.

This allows users to have a guideline as to how the app functions,
and the streamlined help command provides in-app guidance to them
regarding the different commands they are able to use in the app.

phiphi-tan committed:
Add UG and make GUI changes

phiphi-tan committed:
Update README.md

phiphi-tan committed:
make changes to UG to fit HTML conversion

phiphi-tan committed:
Attempt to fix table display in github pages

phiphi-tan committed:
Remove section links in table for UG format

phiphi-tan committed:
Used HTML tags instead of GFMD format for table

phiphi-tan committed:
Experiment with different types of markdown

Table on github pages is not displaying correctly even though
the preview on markdown is accurate.

This is most likely due to the markdown to HTML conversion

phiphi-tan committed:
Remove table header seperator
